site/docs/latest/admin/Upgrade.md

```diff
@@ -0,0 +1,152 @@
+- Always upgrade one datacenter to canry new version before upgrading all datacenters if your cluster is running in multi-clusters replicated mode.
+
+Following is the general guide to upgrade an Apache Pulsar cluster. If there is anything specials needed to be taken care, we will describe the details
+for individual versions in "Upgrade Guides" section.
+
+## Sequence
+
+Follow the sequence below on upgrading an Apache Pulsar cluster
+
+1. Upgrade Bookies
+  1. Canary: canary an upgraded version in one or small set of bookies.
+  2. Rolling Upgrade: rollout the upgraded version to all bookies in the cluster once you determined a version is safe after canary.
+2. Upgrade Brokers
+  1. Canary: canary an upgraded version in one or small set of brokers.
+  2. Rolling Upgrade: rollout the upgraded version to all brokers in the cluster once you determined a version is safe after canary.
+3. Upgrade Proxies
+  1. Canary: canary an upgraded version in one or small set of proxies.
+  2. Rolling Upgrade: rollout the upgraded version to all proxies in the cluster once you determined a version is safe after canary.
+
+Typically you don't need to upgrade zookeeper unless there are strong reasons documented in "Upgrade Guides" section below.
+But if you would like to keep your zookeeper cluster consistent with other components, follows the general upgrade guideline as others.
+
+1. Canary: canary an upgraded version in one of the zookeeper servers.
+2. Rolling Upgrade: rollout the upgraded version to all zookeeper servers incrementally, one at a time.
+
+## Upgrade Bookies
+
+> You can also read Apache BookKeeper [Upgrade Guide](http://bookkeeper.apache.org/docs/latest/admin/upgrade) for more details.
+
+### Canary
+
+It is wise to canary an upgraded version in one or small set of bookies before upgrading all bookies in your live cluster.
+
+You can follow below steps on how to canary a upgraded version:
+
+1. Stop a Bookie.
+2. Upgrade the binary and configuration.
+3. Start the Bookie in `ReadOnly` mode. This can be used to verify if the Bookie of this new version can run well for read workload.
+   ```shell
+   bin/pulsar bookie --readOnly
+   ```
+4. Once the Bookie is running at `ReadOnly` mode successfully for a while, restart the Bookie in `Write/Read` mode.
+   ```shell
+   bin/pulsar bookie
+   ```
+5. After step 4, the Bookie will serve both write and read traffic.
+
+#### Rollback Canaries
+
+If problems occur during canarying an upgraded version, you can simply take down the problematic Bookie node. The remain bookies in the old cluster
+will repair this problematic bookie node by autorecovery. Nothing needs to be worried about.
+
+### Upgrade Steps
+
+Once you determined a version is safe to upgrade in a few bookies in your cluster, you can perform following steps to upgrade all bookies in your cluster.
+
+1. Decide on performing a rolling upgrade or a downtime upgrade.
```

Comment:
is "downtime upgrade" common terminology. I understand what it means, but there's probably a better way to phrase it.

"Decide whether you want to upgrade the whole cluster at once, implying downtime, or do a rolling upgrade."